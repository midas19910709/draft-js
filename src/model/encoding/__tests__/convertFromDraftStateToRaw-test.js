/**
 * Copyright (c) 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @emails oncall+ui_infra
 * @format
 * @flow
 */

'use strict';

jest.disableAutomock();

const BlockMapBuilder = require('BlockMapBuilder');
const ContentBlockNode = require('ContentBlockNode');
const Immutable = require('immutable');

const convertFromDraftStateToRaw = require('convertFromDraftStateToRaw');
const getSampleStateForTesting = require('getSampleStateForTesting');

const {contentState} = getSampleStateForTesting();

const treeContentState = contentState.set(
  'blockMap',
  BlockMapBuilder.createFromArray([
    new ContentBlockNode({
      key: 'A',
      children: Immutable.List.of('B', 'E'),
    }),
    new ContentBlockNode({
      parent: 'A',
      key: 'B',
      nextSibling: 'C',
      children: Immutable.List.of('C', 'D'),
    }),
    new ContentBlockNode({
      parent: 'B',
      key: 'C',
      text: 'left block',
      nextSibling: 'D',
    }),
    new ContentBlockNode({
      parent: 'B',
      key: 'D',
      text: 'right block',
      prevSibling: 'C',
    }),
    new ContentBlockNode({
      parent: 'A',
      key: 'E',
      text: 'This is a tree based document!',
      type: 'header-one',
      prevSibling: 'B',
    }),
  ]),
);

const assertConvertFromDraftStateToRaw = content => {
  expect(convertFromDraftStateToRaw(content)).toMatchSnapshot();
};

test('must be able to convert from draft state with ContentBlock to raw', () => {
  assertConvertFromDraftStateToRaw(contentState);
});

test('must be able to convert from draft state with ContentBlockNode to raw', () => {
  assertConvertFromDraftStateToRaw(treeContentState);
});
