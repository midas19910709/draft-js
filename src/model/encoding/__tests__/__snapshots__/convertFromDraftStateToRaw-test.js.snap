// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must be able to convert from draft state with ContentBlock to raw 1`] = `
Object {
  "blocks": Array [
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [],
      "key": "a",
      "text": "Alpha",
      "type": "unstyled",
    },
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [
        Object {
          "key": 0,
          "length": 5,
          "offset": 0,
        },
      ],
      "inlineStyleRanges": Array [
        Object {
          "length": 5,
          "offset": 0,
          "style": "BOLD",
        },
      ],
      "key": "b",
      "text": "Bravo",
      "type": "unordered-list-item",
    },
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [],
      "key": "c",
      "text": "Test",
      "type": "code-block",
    },
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [],
      "key": "d",
      "text": "",
      "type": "code-block",
    },
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [],
      "key": "e",
      "text": "",
      "type": "code-block",
    },
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [
        Object {
          "length": 7,
          "offset": 0,
          "style": "ITALIC",
        },
      ],
      "key": "f",
      "text": "Charlie",
      "type": "blockquote",
    },
  ],
  "entityMap": Object {
    "0": Object {
      "data": Object {},
      "mutability": undefined,
      "type": Object {
        "data": null,
        "mutability": "IMMUTABLE",
        "type": "IMAGE",
      },
    },
  },
}
`;

exports[`must be able to convert from draft state with ContentBlockNode to raw 1`] = `
Object {
  "blocks": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "data": Object {},
              "depth": 0,
              "entityRanges": Array [],
              "inlineStyleRanges": Array [],
              "key": "C",
              "text": "left block",
              "type": "unstyled",
            },
            Object {
              "children": Array [],
              "data": Object {},
              "depth": 0,
              "entityRanges": Array [],
              "inlineStyleRanges": Array [],
              "key": "D",
              "text": "right block",
              "type": "unstyled",
            },
          ],
          "data": Object {},
          "depth": 0,
          "entityRanges": Array [],
          "inlineStyleRanges": Array [],
          "key": "B",
          "text": "",
          "type": "unstyled",
        },
        Object {
          "children": Array [],
          "data": Object {},
          "depth": 0,
          "entityRanges": Array [],
          "inlineStyleRanges": Array [],
          "key": "E",
          "text": "This is a tree based document!",
          "type": "header-one",
        },
      ],
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [],
      "key": "A",
      "text": "",
      "type": "unstyled",
    },
  ],
  "entityMap": Object {},
}
`;
